### DISCUSSION TITLE
Does pause also pause wireguard?

### DISCUSSION THREAD

**Discussion by iulko** [celzero/rethink-app#1286] [General]
Does pausing also pause wireguard/proxy connection? 
I don't know what should be the right way to do it, just wanted to know so we can discuss what should happen.

**Comment by ignoramous** [comment]
>  I don't know what should be the right way to do it, just wanted to know so we can discuss what should happen.

Pause does not pause WireGuard.

1. It removes all "allowed" apps from Rethink's VPN tunnel; "blocked" apps remain part of the tunnel and hence subject to all of the per-app / universal (global) IP & domain rules.
2. If VPN is in *Lockdown* mode (`Block connections without VPN` is turned ON) no app is removed from Rethink's VPN tunnel. But: "allowed" apps are exempt from ALL rules (including per-app rules and routing rules, ie proxying) and some domain rules; "blocked" apps are *not* exempt from ANY pre-app rules or firewall rules or routing rules.

Everything else continues to work as-is.
